### PR TITLE
soc: nxp: rw61x: enable support for USB device mode

### PR DESCRIPTION
**Pull request by danieldegrasse**
Enable support for USB device mode on the RW61x SOC. This support was verified on the `rd_rw612_bga` target, using the following samples:

- `samples/subsys/usb/console`
- `samples/subsys/usb/dfu`

### CODE REVIEW

File: drivers/usb/device/usb_dc_mcux.c [hunk 1]
```diff
@@ -95,6 +95,8 @@ BUILD_ASSERT(NUM_INSTS <= 1, "Only one USB device supported");
 #elif DT_NODE_HAS_STATUS(DT_NODELABEL(usb2), okay)
 #define CONTROLLER_ID kUSB_ControllerEhci1
 #endif /* IMX RT */
+#elif defined(CONFIG_SOC_SERIES_RW6XX)
+#define CONTROLLER_ID kUSB_ControllerEhci0
```
Comment (mmahadevan108) [inline drivers/usb/device/usb_dc_mcux.c:99]: Can we avoid adding this by using the nod label `usb1`?

Comment (danieldegrasse) [reply drivers/usb/device/usb_dc_mcux.c:99]: We can't, no- take a look at how the code above depends on `#elif defined(CONFIG_SOC_SERIES_IMXRT11XX) || defined(CONFIG_SOC_SERIES_IMXRT10XX)`. I discussed this with @decsny here: https://github.com/zephyrproject-rtos/zephyr/pull/63999#discussion_r1383676505. I still am of the opinion we should not be introducing SOC specific Kconfigs into a driver unless it is absolutely required, but that is outside the scope of this PR.

Comment (mmahadevan108) [reply drivers/usb/device/usb_dc_mcux.c:99]: Makes sense. No point spending time restructuring this driver since we are moving to a new driver.

**Review by mmahadevan108 — DISMISSED** [review]

**Review by decsny — DISMISSED** [review]

**Comment by dleach02** [comment]
@jfischer-no, we need your review/approval on this PR

**Comment by danieldegrasse** [comment]
Rebased to clear merge conflicts

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by tmon-nordic — APPROVED** [review]

**Comment by dleach02** [comment]
@jfischer-no, I've switched my name in the assignee. Please switch back if you are prepared to review.

**Pull request merged**